src/main/scala/com/mozilla/telemetry/views/CrashSummaryView.scala

```diff
@@ -229,14 +229,15 @@ object CrashSummaryView extends BatchJobBase {
       val processedPings = spark.sparkContext.longAccumulator("processedPings")
       val discardedPings = spark.sparkContext.longAccumulator("discardedPings")
       val crashPings = messages.records()
-        .map(x => this.transformPayload(x.fieldsAsMap, x.payload))
-      crashPings.foreach(x => {
-        if (!x.isDefined) {
-          discardedPings.add(1)
-        } else {
-          processedPings.add(1)
-        }
-      })
+        .map(x => {
+          val payload = this.transformPayload(x.fieldsAsMap, x.payload);
+          if (!payload.isDefined) {
```

Comment:
```suggestion
          if (payload.isEmpty) {
```
^would be more idiomatic.